hack/scripts/provision.sh

```diff
@@ -294,20 +295,38 @@ install_flintlockd() {
 
 # Fetch and start the flintlock systemd service
 start_flintlockd_service() {
+	local address="$1"
+
 	say "Starting flintlockd service"
 
 	service=$(basename "$FLINTLOCKD_SERVICE_FILE")
 	fetch_service_file "$FLINTLOCK_REPO" "$service" "$FLINTLOCKD_SERVICE_FILE"
+	edit_service_file "$address"
+	start_service "$FLINTLOCK_BIN"
+
+	say "Flintlockd running at $address:9090"
+}
+
+# This is a temporary work-around func while I sort out a config file for
+# flintlock
+# Don't look too closely at it, it is not long for this world :p
+edit_service_file() {
+	local address="$1"
 
-	# these are temporary while I sort out a configfile for flintlockd
 	parent=$(ip route show | awk '/default/ {print $5}')
 	sed -i "s/PARENT_IFACE/$parent/" "$FLINTLOCKD_SERVICE_FILE"
 	socket="$CONTAINERD_STATE_DIR/containerd.sock"
 	sed -i "s|\(socket=\).*\(\\)|\1$socket \\\\\2|g" "$FLINTLOCKD_SERVICE_FILE"
 
-	start_service "$FLINTLOCK_BIN"
+	if [[ -z "$address" ]]; then
+		address=$(lookup_address)
+	fi
+	sed -i "s/ADDRESS/$address/" "$FLINTLOCKD_SERVICE_FILE"
```

Comment:
I think that `-i` can cause trouble, if it does not, it will.

Reply:
yeh, a lot of those `-i`s are a prblem, but hopefully i can remove that whole bit soon